### PR TITLE
component generator

### PR DESCRIPTION
**Pull request by michelengelen**
adding component generator and fixing deployment to gh pages

IconButton will be done in another PR

### CODE REVIEW

File: scripts/component-templates.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+import ${name}Base from './${name}.base';
+import {
+    DEFAULT_${toUpper(snakeCase(name))}_SIZE,
+    ${toUpper(snakeCase(name))}_DEFINITIONS,
+} from './${name}.constants';
+import P${name} from './${name}.props';
+
+const ${name} = styled(${name}Base)<P${name}>(
+    ({
+        size = DEFAULT_${toUpper(snakeCase(name))}_SIZE,
```
Comment (nevyangelova) [inline scripts/component-templates.js:23]: just to keep in mind to follow the naming convention, perhaps we should document that later

File: scripts/component-templates.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+);
+
+export default ${name}Base;
+`;
+
+// component.stories.mdx
+const story = (
+    name
+) => `import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
+import { Utils, DEFAULT_ARGUMENTSTABLE_EXCLUSION } from '../../shared';
```
Comment (nevyangelova) [inline scripts/component-templates.js:57]: nit: i know we keep changing that name but can we add the space between the ARGUMENTS_TABLE :D

**Review by nevyangelova — APPROVED** [review]
LG

File: scripts/component-templates.js (hunk 2 above)
Comment (michelengelen) [reply scripts/component-templates.js:57]: I will rename it in my next PR ... 👍🏼

File: scripts/component-templates.js (hunk 1 above)
Comment (michelengelen) [reply scripts/component-templates.js:23]: it does ... and yes ... we should document that.

File: scripts/component-templates.js (hunk 2 above)
Comment (michelengelen) [reply scripts/component-templates.js:57]: done! :D

**Pull request merged**